File: Frontend/src/classes/HealthManager.ts
import HealthForm from "./HealthForm";
import User from "./User";

export default class HealthManager {
    private static instance: HealthManager | null = null;

    private constructor() {

    }

    public static getInstance(): HealthManager {
        if(this.instance == null) {
            this.instance = new HealthManager();
        }
        return this.instance;
    }

    public createForm(user: User, date: string, place: string, time: string): boolean {
        //TO DO
        return true;
    }

    public ambulanceCall(phoneNumber: string): boolean {
        //TO DO
        return true;
    }

    public cancelOrder(healthForm: HealthForm): boolean {
        //TO DO
        return true;
    }
}

File: Backend/src/main/java/com/example/backend/HealthManager.java
package com.example.backend;

public class HealthManager {

    //Properties
    private static HealthManager instance = null;

    //Constructor
    private HealthManager(){}

    //Methods
    public static HealthManager getInstance(){
        if(instance == null) {
            instance = new HealthManager();
        }
        return instance;
    }

    public boolean createForm(User user, String date, String place, String time) {
        //TO DO
        return true;
    }

    public boolean ambulanceCall(String phoneNumber){
        //TO DO
        return true;
    }

    public boolean cancelOrder(HealthForm healthForm){
        //TO DO
        return true;
    }

}
